database/bigquery/store_query_results.py

```diff
@@ -0,0 +1,117 @@
+import os
+import json
+import tempfile
+import argparse
+
+import pandas as pd
+
+from google.cloud import bigquery
+from google.oauth2 import service_account
+from google.api_core.exceptions import NotFound
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--query', dest='query', required=True)
+    parser.add_argument('--service-account', dest='service_account', required=True)
+    parser.add_argument('--destination-file-name', dest='destination_file_name',
+            default='output.csv', required=True)
+    parser.add_argument('--destination-folder-name',
+            dest='destination_folder_name', default='', required=False)
+    args = parser.parse_args()
+    return args
+
+
+def set_environment_variables(args):
+    """
+    Set GCP credentials as environment variables if they're provided via keyword
+    arguments rather than seeded as environment variables. This will override
+    system defaults.
+    """
+    credentials = args.service_account
+    try:
+        json_credentials = json.loads(credentials)
+        fd, path = tempfile.mkstemp()
+        print(f'Storing json credentials temporarily at {path}')
+        with os.fdopen(fd, 'w') as tmp:
+            tmp.write(credentials)
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = path
+        return path
+    except Exception:
+        print('Using specified json credentials file')
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = credentials
+        return
+
+
+def combine_folder_and_file_name(folder_name, file_name):
+    """
+    Combine together the provided folder_name and file_name into one path variable.
+    """
+    combined_name = os.path.normpath(
+        f'{folder_name}{"/" if folder_name else ""}{file_name}')
+
+    return combined_name
+
+
+def create_csv(query, client, destination_file_path):
+    """
+    Read in data from a SQL query. Store the data as a csv.
+    """
+    try:
+        data = client.query(query).to_dataframe()
```

Comment:
I'm assuming this is done in an efficient manner? Is there a need to chunk or anything. There sdk will take care of it?

Reply:
Neither Google's bigquery default API, or their pandas version of it supports chunked downloads, while it does support chunked uploads but that doesn't help us in this case.

Reply:
Confirming that there's no better way to handle this. I built out the "Export to GCS function" to follow the Google preferred method for exporting.